### ISSUE TITLE
Using class bound methods as the RHS for solving ODEs

### ISSUE THREAD

**Issue by spraharsh** [SciML/diffeqpy#73]
I have a function bound to a class that I want to use for solving my ODEs, for example
```python
class potclass:
    def __init__(self):
        self.f_eval = 0
    def get_negative_grad(self, u, p, t):
        self.f_eval += 1
        return -2*u
```
where I'm interested in using `pot.get_negative_grad`  as `f(u, p, t)`

I'm now initializing my problem with

```python
u0 = 0.5
pot = potclass()
tspan = (0, 1.0)
print(pot.get_negative_grad(1, 1, 1))
prob = de.ODEProblem(pot.get_negative_grad,u0, tspan)
de.init(prob, de.Tsit5())
``` 
however, running the code with `python-jl` with Julia 1.5 and python 3.8.1 causes an error, maybe since python implicitly uses the first argument as self(?). 

is there any way I can pass class-bound methods as the functions for the ode solver since all my functions of interest are bound to classes? 

I have attached the python file along with the output I got in the zipped folder
[code_and_output.zip](https://github.com/SciML/diffeqpy/files/5177122/code_and_output.zip)

**Comment by tkf** [comment]
OK. So I think I get a repro https://github.com/SciML/diffeqpy/pull/76#issuecomment-687545738 and a fix #74. Feel free to merge it.

**Comment by ChrisRackauckas** [comment]
Thanks!
